### DISCUSSION TITLE
如何通过一个演示，介绍CVE、钉钉等应用APP的使用？

### DISCUSSION THREAD

**Discussion by wzfukui** [flagify-com/OctoMation#68] [Show and tell]
根据issue #67 中提到的需求，建立该讨论。
请[qqdan3](https://github.com/qqdan3) 描述一下需求内容。

**Comment by qqdan3** [comment]
老师  您好
     实在是抱歉，我不懂编程，所以只是按照老师的教程一步步的模仿，还请您见谅。
 我现在有3个问题 ：
 1、我做了一个CVE的剧本（搜集redhat的CVE,并且进行了等级的要求），然后钉钉也做好配置，但是这个有信息后首先是空的，只有“链接”两个字，或许是没有检索到我要的信息，再有就是没有能推送到钉钉。需要哪里在匹配下么？
 2、我想增加github\cnnvd（国家漏洞库）的漏洞库检测，这个要怎么去做呢？ 有没有剧本可以实现?
 3、我看您介绍资料的例子里还有fortinet的防火墙，这个是用来做啥的？ 这个工具可以被下载到么？
以上，这个编排对我来说却是有点难，还请见谅，也请多指教，谢谢。

**Reply by wzfukui** [reply]
收到，安排起来，会尽快给出演示。

**Comment by wzfukui** [comment]
先来理解一下第一个需求，看不是不是图上的意思：
## 需求一
![image](https://github.com/flagify-com/OctoMation/assets/146505187/1108e5ee-9d98-4c26-bac3-316e8651f044)
这里边暗含了几个逻辑：
- 定时查询CVE（剧本要定时执行）
- 查询CVE带不带关键字符串，例如：Chrome、CentOS、WPS等等？
- CVE漏洞有严重度或者CVSS评分，按照什么分值判定严重？
- 如果是严重漏洞就发送消息到钉钉（提前准备好钉钉机器人）

如果是以上逻辑的话，通过OctoMation实现是没有问题的。您现在使用cve APP查询不到数据是因为Redhat的API近期做了更新，我们这两天就会update逻辑，很快完成适配。

## 需求二
和上一个类似，关键是是否有信息获取源。

## 问题三
Fortigate是Fortinet公司的防火墙名称，与此类似的有华为防火墙、深信服防火墙等等。如果有个安全事件经过剧本逻辑判断，需要封禁，那么可以在剧本中增加一个动作，使用企业内部的防火墙封禁一个IP地址（剧本或者动作的输入）。

社区产品所有的应用包括源码可以通过https://github.com/flagify-com/OctoMationApps进行下载。当前仓库中还没有增加，很快会放上去。

**Comment by wzfukui** [comment]
剧本草稿设计完成，请参考，确认无误的话，咱直播演示。
![WX20231120-220736@2x](https://github.com/flagify-com/OctoMation/assets/146505187/7b74b730-c0ee-46ae-a174-2dd6b09104fe)

**Comment by qqdan3** [comment]
老师 您好
 非常感谢收到您的回复，收集CVE这个，不知道咱们的剧本是能包含哪些网站。 之前我们是从cvnnd.org.cn收集，但是后来改版后就无法获取了，目前已知的是这个国家漏洞库，还有国外的nist、CVE的漏洞库会是比较多。
 然后我想问下，如果导入您说的剧本，后面是有费用的么？ 
 最后想问下，退出来的CVE信息是怎么和钉钉挂钩的？
以上，谢谢老师。 
------------------------------------------------------------------
发件人：fukui ***@***.***>
发送时间：2023年11月20日(星期一) 22:08
收件人：flagify-com/OctoMation ***@***.***>
抄　送：qqdan3 ***@***.***>; Comment ***@***.***>
主　题：Re: [flagify-com/OctoMation] 如何通过一个演示，介绍CVE、钉钉等应用APP的使用？ (Discussion #68)
剧本草稿设计完成，请参考，确认无误的话，咱直播演示。
 <https://user-images.githubusercontent.com/146505187/284290302-7b74b730-c0ee-46ae-a174-2dd6b09104fe.png >

**Comment by wzfukui** [comment]
@qqdan3 参考这个视频： 2.07 Duf:/ 08/13 F@u.sE 安全编排自动化：CVE漏洞查询与通告（上）——OctoMation手把手视频系列 # SOAR # OctoMation # 安全自动化 # 编排自动化 # 安全低代码  https://v.douyin.com/iRpYTqt1/ 复制此链接，打开Dou音搜索，直接观看视频！

- 数据来源API推荐使用：https://nvd.nist.gov/developers/vulnerabilities
- 如果导入您说的剧本，后面是有费用的么？：剧本和应用APP都是免费的
- 最后想问下，退出来的CVE信息是怎么和钉钉挂钩的？ ： 看视频

**Comment by qqdan3** [comment]
老师 您好
 因为最近单位有些检查事项，所以没能及时测试，也没能给您回复，实在抱歉。
 同时，感谢您给我们的支持，谢谢。
 我看了视频，然后再操作是，看到用例中的 NVD的CVE的APP，这个从哪里下载？ 我只有CVE search 这个，目前使用CVE search 找不到相关的点，所以没能跟上操作。
 所以能给我一个这个APP的下载地址么？ 谢谢。
------------------------------------------------------------------
发件人：fukui ***@***.***>
发送时间：2023年11月26日(星期日) 12:52
收件人：flagify-com/OctoMation ***@***.***>
抄　送：qqdan3 ***@***.***>; Mention ***@***.***>
主　题：Re: [flagify-com/OctoMation] 如何通过一个演示，介绍CVE、钉钉等应用APP的使用？ (Discussion #68)
@qqdan3 <https://github.com/qqdan3 > 参考这个视频： 2.07 Duf:/ 08/13 ***@***.*** ***@***.*** > 安全编排自动化：CVE漏洞查询与通告（上）——OctoMation手把手视频系列 # SOAR # OctoMation # 安全自动化 # 编排自动化 # 安全低代码 https://v.douyin.com/iRpYTqt1/ <https://v.douyin.com/iRpYTqt1/ > 复制此链接，打开Dou音搜索，直接观看视频！

 * 数据来源API推荐使用：https://nvd.nist.gov/developers/vulnerabilities <https://nvd.nist.gov/developers/vulnerabilities >
 * 如果导入您说的剧本，后面是有费用的么？：剧本和应用APP都是免费的
 * 最后想问下，退出来的CVE信息是怎么和钉钉挂钩的？ ： 看视频

**Reply by wuzhi-dev** [reply]
https://github.com/flagify-com/OctoMationApps/tree/main/shakespeare-action-python-cve_search  下载这个应用代码然后打成zip包通过系统中的应用管理上传即可

**Reply by wuzhi-dev** [reply]
https://github.com/flagify-com/OctoMationApps/releases/download/APP/shakespeare-action-python-cve_search.zip  这个链接更新了 也可以直接下载使用

**Comment by qqdan3** [comment]
谢谢老师，可以下载，我继续测试下。
------------------------------------------------------------------
发件人：wuzhi-dev ***@***.***>
发送时间：2023年11月30日(星期四) 15:47
收件人：flagify-com/OctoMation ***@***.***>
抄　送：qqdan3 ***@***.***>; Mention ***@***.***>
主　题：Re: [flagify-com/OctoMation] 如何通过一个演示，介绍CVE、钉钉等应用APP的使用？ (Discussion #68)
https://github.com/flagify-com/OctoMationApps/releases/download/APP/shakespeare-action-python-cve_search.zip <https://github.com/flagify-com/OctoMationApps/releases/download/APP/shakespeare-action-python-cve_search.zip > 这个链接更新了 也可以直接下载使用

**Comment by qqdan3** [comment]
老师 您好
 您给我的连接是cve search ，演示中的例子是用nvd cve ， 请问有后者的APP 包么，我之前装过cve searche这个APP，谢谢老师
------------------------------------------------------------------
发件人：wuzhi-dev ***@***.***>
发送时间：2023年11月30日(星期四) 15:47
收件人：flagify-com/OctoMation ***@***.***>
抄　送：qqdan3 ***@***.***>; Mention ***@***.***>
主　题：Re: [flagify-com/OctoMation] 如何通过一个演示，介绍CVE、钉钉等应用APP的使用？ (Discussion #68)
https://github.com/flagify-com/OctoMationApps/releases/download/APP/shakespeare-action-python-cve_search.zip <https://github.com/flagify-com/OctoMationApps/releases/download/APP/shakespeare-action-python-cve_search.zip > 这个链接更新了 也可以直接下载使用

**Reply by wuzhi-dev** [reply]
https://github.com/flagify-com/OctoMationApps/releases/download/APP/shakespeare-action-python-nvd_cve.zip  这个是nvd的 试试

**Comment by qqdan3** [comment]
谢谢老师，信息收到
 我在按照视频教程时，请问在第一部添加APP后，要求写入关键字，然后视频中是自定义test1,，请问这个自建库从哪里开始建立？ 一般的话这个都是要手动输入关键字的吧？
 另外，在第二部设置条件时，需要写入出参-基础得分basic score，但是我的环境中找不到这项。是不是我哪里选择不对呢？ 谢谢
------------------------------------------------------------------
发件人：wuzhi-dev ***@***.***>
发送时间：2023年11月30日(星期四) 17:20
收件人：flagify-com/OctoMation ***@***.***>
抄　送：qqdan3 ***@***.***>; Mention ***@***.***>
主　题：Re: [flagify-com/OctoMation] 如何通过一个演示，介绍CVE、钉钉等应用APP的使用？ (Discussion #68)
https://github.com/flagify-com/OctoMationApps/releases/download/APP/shakespeare-action-python-nvd_cve.zip <https://github.com/flagify-com/OctoMationApps/releases/download/APP/shakespeare-action-python-nvd_cve.zip > 这个是nvd的 试试

**Reply by wuzhi-dev** [reply]
视频中的应用之前还在pre-release，目前已经完成发布并更新到库里了，老师可以用上面的连接下载更新，谢谢。

**Comment by qqdan3** [comment]
老师 好
 信息收到，谢谢。
 我随后更新下试试，有问题再咨询您，谢谢。
------------------------------------------------------------------
发件人：wuzhi-dev ***@***.***>
发送时间：2023年12月1日(星期五) 10:45
收件人：flagify-com/OctoMation ***@***.***>
抄　送：qqdan3 ***@***.***>; Mention ***@***.***>
主　题：Re: [flagify-com/OctoMation] 如何通过一个演示，介绍CVE、钉钉等应用APP的使用？ (Discussion #68)
视频中的应用之前还在pre-release，目前已经完成发布并更新到库里了，老师可以用上面的连接下载更新，谢谢。
